### ISSUE TITLE
Remove internal dependencies on web3.eth.contract

### ISSUE THREAD

**Issue by benjamincburns** [trufflesuite/truffle#5475]
Instead of internally depending on `web3.eth.contract`, we should instead depend on `@truffle/contract`. This would remove a great deal of our internal dependence on `web3.js`, and it would cause us to better define `@truffle/contract` for the needs of the Truffle project as a whole.

Creating this as an (empty) epic, as it's likely that a fair amount of scoping work would need to be done before we can embark on this effort.

Note: This was captured from an internal session where we discussed **potential** future breaking changes for major releases. Until the team assigns this ticket, this is merely a proposal, and not a commitment to make this specific change.

**Comment by haltman-at** [comment]
This should probably be combined with #5474; as far as I'm aware, the only place we rely on web3.eth.Contract is inside Truffle Contract.

**Comment by benjamincburns** [comment]
Yeah, I agree. Closing in favor of that one.
